File: Java-Hospital-SOAP/src/main/java/ru/hazov/javahospitalsoap/service/DiseaseService.java
package ru.hazov.javahospitalsoap.service;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import ru.hazov.javahospitalsoap.entity.Disease;
import ru.hazov.javahospitalsoap.jaxb_model.CreateDiseaseRequest;
import ru.hazov.javahospitalsoap.repository.DiseaseRepository;


@Service
public class DiseaseService {

    private final DiseaseRepository diseaseRepository;

    @Autowired
    public DiseaseService(DiseaseRepository diseaseRepository) {
        this.diseaseRepository = diseaseRepository;
    }

    public void createDisease(CreateDiseaseRequest request) {
        Disease disease = new Disease();
        disease.setName(request.getName());
        disease.setDescription(request.getDescription());
        diseaseRepository.save(disease);
    }

    public Disease getDiseaseByName(String diseaseName) {
        return diseaseRepository.findByName(diseaseName);
    }
}

File: CSharpHospitalREST/CSharpHospitalREST/Service/DiseaseService.cs
using CSharpHospitalREST.config.db;
using CSharpHospitalREST.dto.disease.CreateDisease;
using CSharpHospitalREST.Models;

namespace CSharpHospitalREST.Service;

public class DiseaseService(HospitalDatabaseContext db)
{
    public Disease? GetDiseaseByName(string diseaseName)
    {
        return db.Diseases.First(d => d.Name.Equals(diseaseName));
    }
    
    public Disease CreateDisease(CreateDiseaseRequest request)
    {
        var disease = new Disease
        {
            Name = request.Name,
            Description = request.Description,
        };
        db.Diseases.Add(disease);
        return disease;
    }
    
}